addons/pos_self_order_mollie/models/pos_payment_method.py

```diff
@@ -0,0 +1,59 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from werkzeug import urls
+
+from odoo import models, fields
+
+
+class PosPaymentMethod(models.Model):
+    _inherit = "pos.payment.method"
+
+    def _payment_request_from_kiosk(self, order):
+        if self.use_payment_terminal != 'mollie':
+            return super().payment_request_from_kiosk(order)
+        else:
+            payment_data = {
+                'description': order.name,
+                'pos_reference': order.pos_reference,
+                'currency': order.currency_id.name,
+                'amount': order.amount_total,
+                'session_id': order.session_id.id,
+                'config_id': order.session_id.config_id.id,
+                'order_type': 'kiosk'
+            }
+            result = self.mollie_payment_request(payment_data)
+            return result and result.get('status') == 'open'
+
+    def _get_mollie_webhook_url(self, data):
+        webhook_url = super()._get_mollie_webhook_url(data)
+        if data.get('order_type') == 'kiosk':
+            base_url = self.get_base_url()
+            webhook_url = urls.url_join(base_url, f'/pos_mollie_kiosk/webhook/{self.id}')
+        return webhook_url
```

Comment:
this can be removed after you remove the kiosk webhook

Reply:
Please see details in previous comment